karapace/anonymize_schemas/anonymize_avro.py

```diff
@@ -0,0 +1,145 @@
+"""
+karapace - anonymize avro
+
+Copyright (c) 2022 Aiven Ltd
+See LICENSE for details
+"""
+from typing import Any, Dict, List, Union
+
+import hashlib
+import logging
+
+ALIASES = "aliases"
+DEFAULT = "default"
+DOC = "doc"
+ENUM = "enum"
+FIELDS = "fields"
+ITEMS = "items"
+NAME = "name"
+NAMESPACE = "namespace"
+ORDER = "order"
+SIZE = "size"
+SYMBOLS = "symbols"
+TYPE = "type"
+
+PRIMITIVE_TYPES = ["null", "boolean", "int", "long", "float", "double", "bytes", "string"]
+
+
+def anonymize_name(name: str) -> str:
+    """Anonymize the name.
+
+    Name is splitted by dot to logical elements that the whole name consists of.
+    The element is sha1 hashed. After hashing element is cleaned to conform to Avro name
+    format of:
+      * start with [A-Za-z_]
+      * subsequently contain only [A-Za-z0-9_]
+
+    Returns anonymized name.
+    """
+    anonymized_elements = []
+    for element in name.split("."):
+        element = hashlib.sha1(element.encode("utf-8")).hexdigest()
+        # SHA-1 can start with number, just add 'a' as first character.
+        # This breaks the hash, but is still consistent as required.
+        as_list_element = list(element)
+        as_list_element[0] = "a"
+        anonymized_elements.append("".join(as_list_element))
+
+    return ".".join(anonymized_elements)
+
+
+def anonymize_type(maybe_type_list_or_str: Union[str, List[Any]]) -> Union[str, List[Any]]:
+    if isinstance(maybe_type_list_or_str, list):
+        union_types = []
+        for union_type in maybe_type_list_or_str:
+            if union_type in PRIMITIVE_TYPES:
+                union_types.append(union_type)
+            elif isinstance(union_type, str):
+                union_types.append(anonymize_name(union_type))
+            else:
+                union_types.append(anonymize(union_type))
+        return union_types
+    return maybe_type_list_or_str
+
+
+def anonymize_schema(type_element: Union[str, Dict[str, str]], input_schema: Dict[str, Any]) -> Dict[str, Any]:
+    schema: Dict[str, Any] = {}
+    for key, value in input_schema.items():
```

Comment:
This loop is not handling [logical types](https://avro.apache.org/docs/current/spec.html#Logical+Types).

Reply:
Done.